pkg/config/analysis/analyzers/virtualservice/destinationrules.go

```diff
@@ -116,10 +120,18 @@ func initDestHostsAndSubsets(ctx analysis.Context) map[hostAndSubset]bool {
 	ctx.ForEach(collections.IstioNetworkingV1Alpha3Destinationrules.Name(), func(r *resource.Instance) bool {
 		dr := r.Message.(*v1alpha3.DestinationRule)
 		drNamespace := r.Metadata.FullName.Namespace
-
+		drHost := dr.GetHost()
+		if drNamespace == util.IstioSystemNamespace && drHost != "" && host.IsShortNameForHost(drHost) {
+			// destionation rule's host can not be a short name if destination rule's namespace is "istio-system"
+			m := msg.NewDestinationRuleHostShouldNotBeShortName(r, r.Metadata.FullName.String(), drHost)
+			ctx.Report(collections.IstioNetworkingV1Alpha3Destinationrules.Name(), m)
+		}
+		// Convert the short name host to FQDN, such as convert "review" to "review.default.svc.cluster.local"
+		// using destination rule's namespace otherwise
+		drHost = util.ConvertHostToFQDN(drNamespace, drHost)
```

Comment:
can we add a test for this case?

Reply:
NP, I will add one test case for it.

Reply:
I meant for the original issue. To make sure this PR fixes it